From 21d62face41ffd9e858108f2d45a9abe50ba0db3 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Mon, 19 Nov 2018 14:00:56 +0100
Subject: actually initialize kcrash properly

Summary:
to successfully use kcrash when linking with as-needed (which is a default
flag on many linux distros) one also needs to call KCrash::initialize.

without this kcrash handling is not available for most users.

https://markmail.org/thread/zv5pheijaze72bzs

Reviewers: mludwig

Reviewed By: mludwig

Subscribers: #kile

Tags: #kile

Differential Revision: https://phabricator.kde.org/D17011
---
 src/main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index 3b02716..bd7fef6 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -25,6 +25,7 @@
 #include <QUrl>
 
 #include <KAboutData>
+#include <KCrash>
 #include <KDBusService>
 #include <KEncodingProber>
 #include <KLocalizedString>
@@ -128,6 +129,8 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
 
     KAboutData::setApplicationData(aboutData);
 
+    KCrash::initialize();
+
     app.setApplicationDisplayName(aboutData.displayName());
     app.setOrganizationDomain(aboutData.organizationDomain());
     app.setApplicationVersion(aboutData.version());
-- 
cgit v1.1
